### ISSUE TITLE
Reproduce results of IROS paper

### ISSUE THREAD

**Issue by maxjaritz** [PRBonn/lidar_transfer#4]
Hi,
Thanks for your interesting work and the provided code!

I'd like to reproduce your results from your IROS paper. However, I struggle to understand if all elements are actually available in this repo. Could you provide the steps that are necessary to re-sample the SemanticKitti dataset (64 layers) with the nuScenes sensor setup (32 layers)? I believe that the target config file is missing.

Best,
Max

**Comment by jbehley** [comment]
Hi @maxjaritz,

as far as I know are the files included in the `minimal.zip`, the files you need. There is a `target.yaml` file with the a Velodyne 32 beam configuration. Regarding the different "modes", you have to adapt the `lidar_transfer.yaml`.

I hope that helps.

If you want to reproduce our results on nuScenes, you will also need our annotations, which you can download here:

http://semantic-kitti.org/data/nuscenes-0103-labels.zip

The format is the same as the SemanticKITTI dataset and it uses the same
classes.

Hope that helps.

**Comment by maxjaritz** [comment]
Thanks a lot for the explanation!

**Comment by jbehley** [comment]
If you have any other issues, let me know.

**Comment by bjajoh** [comment]
@jbehley 
I'm also interested in reproducing the work.
Is the actual training pipeline public?
I was not able to find it.
![image](https://user-images.githubusercontent.com/42101377/134486237-2ecdf642-bb60-4b39-bf1a-a8276bd4c118.png)

**Comment by jbehley** [comment]
@bjajoh, when you generated the point clouds from SemanticKITTI that resembles
the appearance from nuScenes or any other dataset, you can use our
RangeNet++ implementation:

https://github.com/PRBonn/lidar-bonnetal

to train RangeNet++ with a modified configuration file (see
https://github.com/PRBonn/lidar-bonnetal/blob/master/train/tasks/semantic/config/arch/darknet53.yaml),
where you have to adapt the sensor parameters to the specifics of
nuScenes or other datasets.

Regarding our experiments reported in the paper using nuScenes:

In the lidar_transfer repository are also scripts (see
auxiliary/convert/) that convert the nuScenes data to the KITTI format.
We labeled the converted point clouds and then compared the RangeNet++
(trained with SemanticKITTI labels transferred to nuScenes) with the
converted nuScenes data.

You can find our labels for scene 103 from nuScenes at

http://semantic-kitti.org/data/nuscenes-0103-labels.zip

These label files use the binary format from SemanticKITTI and provide
for each point a label for the converted data.

For the correlation alignment, Ferdinand adopted the implementation from 

https://github.com/pmorerio/minimal-entropy-correlation-alignment

where he used the "log_coral_loss" with pytorch. 


If you have further questions, don't hesitate to contact me or simply open a new issue.

**Comment by bjajoh** [comment]
@jbehley 
Thank you very much for the detailed information!
